logentries/utils.py

```diff
@@ -178,18 +173,20 @@ def flush(self):
     def emit(self, record):
         # Reset stdout. See: http://docs.python.org/2/library/sys.html#sys.__stdout__
         sys.stdout = sys.__stdout__
-        lock.acquire()
-        try:
-            if not self._started and self.good_config:
-                dbg("Starting Logentries Asynchronous Socket Appender")
+        if self.good_config and not self._thread.is_alive():
+            try:
                 self._thread.start()
+            except RuntimeError: # It's already started.
```

Comment:
This lock is causing emit to be synchronised globally which is ineficient because python checks this for us in threading.

Because the lock is only to avoid double-starting the thread, we can simply try to start it and catch the exception.